libvast/src/taxonomies.cpp

```diff
@@ -25,6 +26,46 @@
 
 namespace vast {
 
+static caf::error to_concepts_impl(const data& d, concepts_type& out) {
+  if (const auto& tl = caf::get_if<list>(&d)) {
+    for (const auto& item : *tl) {
+      if (const auto& x = caf::get_if<record>(&item)) {
+        auto n = x->find("concept");
+        if (n == x->end())
+          continue;
+        if (const auto& c = caf::get_if<record>(&n->second)) {
```

Comment:
I think it would be nice to have a separate function that starts here, i.e., a `data` that is supposed to be a concept:

```cpp
caf::error convert(const data&, concept_type&);
```

That could then be used as a sub-routine in this function (e.g., `to<concept_type>(x)`. At that point, it is a conversion. 😉